app/widgets/trackProcessing/trackProcessingWidget.php

```diff
@@ -77,15 +77,23 @@ public function renderWidget($ps_widget_id, &$pa_settings) {
 			$this->opo_view->setVar('hours', $pa_settings['hours']);
 			
 			$vo_tq = new TaskQueue();
+			$qr_tq_total_count = $this->opo_db->query("
+				SELECT count(*) as _count
+				FROM ca_task_queue ctq 
+				WHERE ctq.completed_on > ?
+			", time() - (60*60*$pa_settings['hours']));
+
 			$qr_completed = $this->opo_db->query("
 				SELECT tq.*, u.fname, u.lname 
 				FROM ca_task_queue tq 
 				LEFT JOIN ca_users u ON u.user_id = tq.user_id 
-				WHERE tq.completed_on > ? 
+				WHERE tq.completed_on > ?
 				ORDER BY tq.completed_on desc
+				LIMIT 100
```

Comment:
please use $this->opn_limit which is already defined and configurable (and also defaults to 100).

I find it interesting that this limit is included in the while condition and we still end up with a timeout.